### PR TITLE
views: fix content type check

### LINKED ISSUE #118: views: problem on recognize the content type
**Problem**
Today I found an issue when I was [saving a deposit](https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/views.py#L493).
For some reason, firefox modified the `Content-type` from simply `application/json` to `application/json; charset=UTF-8`.
And the server wrongly interpreted it and returned error `415`.
With @slint we checked on `werkzeug` and found that the value inside `request.content_type` is the raw value and the correct one should be [request.mimetype](https://github.com/pallets/werkzeug/blob/master/werkzeug/wrappers.py#L1717):

> Like content_type, but without parameters (eg, without charset, type etc.) and always lowercase. For example if the content type is `text/HTML; charset=utf-8` the mimetype would be `text/html`.

See [here](http://werkzeug.pocoo.org/docs/0.11/wrappers/#werkzeug.wrappers.CommonRequestDescriptorsMixin.mimetype).

**Solution**
Replace [this line](https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/views.py#L493):
```python
if request.content_type not in self.loaders:
```
With this:

```python
if request.mimetype not in self.loaders:
```
------------------------------------------------------------------
cc @jirikuncar @lnielsen

### PR DESCRIPTION
**Pull request by hachreak**
* Uses the request mimetype instead of raw content type header.
  (closes #118)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

**Comment by jirikuncar** [comment]
@hachreak can you please add some specific test case?

**Comment by hachreak** [comment]
depends on https://github.com/inveniosoftware/invenio-rest/pull/60

**Comment by hachreak** [comment]
@jirikuncar :shipit: ?

File: tests/test_views_item_patch.py [hunk 1]
```diff
@@ -29,16 +29,20 @@
...
 ]
 
 
-def test_valid_patch(app, test_records, test_patch):
+@pytest.mark.parametrize('content_type', [
+    'application/json-patch+json', 'application/json-patch+json;charset=utf-8'
+])
+def test_valid_patch(app, test_records, test_patch, content_type):
     """Test VALID record patch request (PATCH .../records/<record_id>)."""
+    HEADERS.append(('Content-Type', content_type))
```
Comment (jirikuncar) [inline tests/test_views_item_patch.py]: be careful about modifying mutable global variable

File: tests/test_views_item_patch.py [hunk 2]
```diff
@@ -57,8 +61,12 @@ def test_valid_patch(app, test_records, test_patch):
         assert previous_year != get_json(client.get(url))['metadata']['year']
 
 
-def test_patch_deleted(app, test_records, test_patch):
+@pytest.mark.parametrize('content_type', [
+    'application/json', 'application/json-patch+json;charset=utf-8'
+])
+def test_patch_deleted(app, test_records, test_patch, content_type):
     """Test patching deleted record."""
+    HEADERS.append(('Content-Type', content_type))
```
Comment (jirikuncar) [inline tests/test_views_item_patch.py]: be careful about modifying mutable global variable

File: tests/test_views_item_patch.py [hunk 3]
```diff
@@ -71,8 +79,15 @@ def test_patch_deleted(app, test_records, test_patch):
         assert res.status_code == 410
 
 
-def test_invalid_patch(app, test_records, test_patch):
+@pytest.mark.parametrize('charset', [
+    '', ';charset=utf-8'
+])
+def test_invalid_patch(app, test_records, test_patch, charset):
     """Test INVALID record put request (PUT .../records/<record_id>)."""
+    HEADERS.append(
```
Comment (jirikuncar) [inline tests/test_views_item_patch.py]: be careful about modifying mutable global variable

File: tests/test_views_item_patch.py [hunk 4]
```diff
@@ -111,16 +127,21 @@ def test_invalid_patch(app, test_records, test_patch):
...
 
 
 @mock.patch('invenio_records.api.Record.validate', _mock_validate_fail)
-def test_validation_error(app, test_records, test_patch):
+@pytest.mark.parametrize('content_type', [
+    'application/json', 'application/json-patch+json;charset=utf-8'
+])
+def test_validation_error(app, test_records, test_patch, content_type):
     """Test VALID record patch request (PATCH .../records/<record_id>)."""
+    HEADERS.append(('Content-Type', content_type))
```
Comment (jirikuncar) [inline tests/test_views_item_patch.py]: be careful about modifying mutable global variable

File: tests/test_views_item_put.py [hunk 5]
```diff
@@ -29,16 +29,21 @@
...
 ]
 
 
-def test_valid_put(app, test_records):
+@pytest.mark.parametrize('content_type', [
+    'application/json', 'application/json;charset=utf-8'
+])
+def test_valid_put(app, test_records, content_type):
     """Test VALID record patch request (PATCH .../records/<record_id>)."""
+    HEADERS.append(('Content-Type', content_type))
```
Comment (jirikuncar) [inline tests/test_views_item_put.py]: be careful about modifying mutable global variable

File: tests/test_views_item_put.py [hunk 6]
```diff
@@ -75,8 +85,13 @@ def test_valid_put_etag(app, test_records):
         assert get_json(client.get(url))['metadata']['year'] == 1234
 
 
-def test_put_on_deleted(app, test_records):
+@pytest.mark.parametrize('content_type', [
+    'application/json', 'application/json;charset=utf-8'
+])
+def test_put_on_deleted(app, test_records, content_type):
     """Test putting to a deleted record."""
+    HEADERS.append(('Content-Type', content_type))
```
Comment (jirikuncar) [inline tests/test_views_item_put.py]: be careful about modifying mutable global variable

File: tests/test_views_item_put.py [hunk 7]
```diff
@@ -88,8 +103,16 @@ def test_put_on_deleted(app, test_records):
         assert res.status_code == 410
 
 
-def test_invalid_put(app, test_records):
+@pytest.mark.parametrize('charset', [
+    '', ';charset=utf-8'
+])
+def test_invalid_put(app, test_records, charset):
     """Test INVALID record put request (PUT .../records/<record_id>)."""
+    HEADERS.append(
```
Comment (jirikuncar) [inline tests/test_views_item_put.py]: be careful about modifying mutable global variable

File: tests/test_views_item_put.py [hunk 8]
```diff
@@ -123,14 +146,20 @@ def test_invalid_put(app, test_records):
...
 
 
 @mock.patch('invenio_records.api.Record.validate', _mock_validate_fail)
-def test_validation_error(app, test_records):
+@pytest.mark.parametrize('content_type', [
+    'application/json', 'application/json;charset=utf-8'
+])
+def test_validation_error(app, test_records, content_type):
     """Test when record validation fail."""
+    HEADERS.append(('Content-Type', content_type))
```
Comment (jirikuncar) [inline tests/test_views_item_put.py]: be careful about modifying mutable global variable

File: tests/test_views_list_post.py [hunk 9]
```diff
@@ -35,14 +35,18 @@
...
 
 
-def test_valid_create(app, db, test_data, search_url):
+@pytest.mark.parametrize('content_type', [
+    'application/json', 'application/json;charset=utf-8'
+])
+def test_valid_create(app, db, test_data, search_url, content_type):
     """Test VALID record creation request (POST .../records/)."""
     with app.test_client() as client:
+        HEADERS.append(('Content-Type', content_type))
```
Comment (jirikuncar) [inline tests/test_views_list_post.py]: be careful about modifying mutable global variable

File: tests/test_views_list_post.py [hunk 10]
```diff
@@ -73,9 +77,14 @@ def test_valid_create(app, db, test_data, search_url):
...
 
 
-def test_invalid_create(app, db, test_data, search_url):
+@pytest.mark.parametrize('content_type', [
+    'application/json', 'application/json;charset=utf-8'
+])
+def test_invalid_create(app, db, test_data, search_url, content_type):
     """Test INVALID record creation request (POST .../records/)."""
     with app.test_client() as client:
+        HEADERS.append(('Content-Type', content_type))
```
Comment (jirikuncar) [inline tests/test_views_list_post.py]: be careful about modifying mutable global variable

File: tests/test_views_list_post.py [hunk 11]
```diff
@@ -113,9 +122,14 @@ def test_invalid_create(app, db, test_data, search_url):
...
 
 @mock.patch('invenio_records.api.Record.validate', _mock_validate_fail)
-def test_validation_error(app, db, test_data, search_url):
+@pytest.mark.parametrize('content_type', [
+    'application/json', 'application/json;charset=utf-8'
+])
+def test_validation_error(app, db, test_data, search_url, content_type):
     """Test when record validation fail."""
     with app.test_client() as client:
+        HEADERS.append(('Content-Type', content_type))
```
Comment (jirikuncar) [inline tests/test_views_list_post.py]: be careful about modifying mutable global variable

**Comment by hachreak** [comment]
@jirikuncar done all.

**Comment by jirikuncar** [comment]
LGTM

**Comment by hachreak** [comment]
@jirikuncar :shipit: ?

**Pull request merged**